src/utils/__tests__/validateENS.test.ts

```diff
@@ -0,0 +1,82 @@
+import validateENS from '../validateENS';
+
+describe('valid names', () => {
+  it('domain', () => {
+    expect(validateENS('lol.eth')).toMatchInlineSnapshot(`
+      Object {
+        "valid": true,
+      }
+    `);
+  });
+
+  it('subdomain', () => {
+    expect(validateENS('super.lol.eth')).toMatchInlineSnapshot(`
+      Object {
+        "valid": true,
+      }
+    `);
+  });
+});
+
+describe('invalid names', () => {
+  it('invalid domain', () => {
+    expect(validateENS('wat')).toMatchInlineSnapshot(`
+      Object {
+        "code": "invalid-domain",
+        "hint": "This is an invalid domain",
+        "valid": false,
+      }
+    `);
+  });
+
+  it('domain with special characters', () => {
+    expect(validateENS('omg$.eth')).toMatchInlineSnapshot(`
+      Object {
+        "code": "invalid-domain-name",
+        "hint": "Your name can not include special characters",
```

Comment:
very nit picky but i feel like can not should be cannot lol

Reply:
also, i noticed that ens names that contain numbers such as ben1.eth give this warning even though they are valid names

Reply:
I have updated to fit these cases and renamed to cannot!